dessia_common/excel_reader.py

```diff
@@ -0,0 +1,482 @@
+
+        Returns:
+        - Depending on the data type of the cell:
+          - For 'Dict' type: Returns a dictionary with keys extracted from the cell
+            and replaced values from the 'values' parameter.
+          - For other types (excluding 'List', 'Set', 'Tuple'): Replaces the values with
+            the first element of 'values' if 'values' is a list and 'cell.value' does not
+            contain any of the container types.
+          - Otherwise, returns 'values' itself.
+        """
+        container = ["Dict", "List", "Set", "Tuple"]
+        target = self.get_location(cell)
+        dict_keys = []
+        if "Dict" in cell.value:
+            real_attr_val = {}
+            sheet = self.workbook[target]
+            column = sheet["A"]
+            for row in column[3:]:
+                dict_keys.append(row.value.split('.')[1])
+
+            for key, value in zip(dict_keys, values):
+                real_attr_val[key] = value
+
+            return real_attr_val
+
+        if isinstance(values, list) and (not any(type in cell.value for type in container)):
+            val_replace = values[0]
+        else:
+            val_replace = values
+
+        return val_replace
+
+    @staticmethod
+    def get_attributes_and_types(obj_class):
+        """
+        Retrieve attributes and their corresponding types from the __init__ method of a class.
+
+        Args:
+        - obj_class: The class object.
+
+        Returns:
+        - dict: A dictionary containing attributes and their respective types defined in the class constructor.
+        """
+        init_signature = inspect.signature(obj_class.__init__)
+        init_parameters = init_signature.parameters
+
+        attribute_info = {}
+
+        for attr_name, param in init_parameters.items():
+            if attr_name != 'self':
+                attr_info = {
+                    'type': param.annotation if param.annotation != param.empty else None,
+                    'default_value': param.default if param.default != param.empty else "empty"
+                }
+                attribute_info[attr_name] = attr_info
+
+        return attribute_info
+
+    @staticmethod
+    def get_data(value, attributes, init_attributes):
+        """
+        Extracts relevant data based on attributes and initial attributes for object instantiation.
+
+        Args:
+        - value (list): List containing values corresponding to attributes.
+        - attributes (list): List of attributes related to the object.
+        - init_attributes (list): List of initial attributes for object instantiation.
+
+        Returns:
+        - dict: Dictionary containing extracted object data based on matching attributes and initial attributes.
+        """
+        check_missed_attribute, attribute = missed_attribute(attributes, init_attributes)
+        if not check_missed_attribute:
+            raise ValueError(f"missing attribute '{attribute}' don't have a default value")
```

Comment:
As "missing attribute 'attribute'" is 3rd person singular, it should be "does not" instead of "don't".

Moreover, please capitalize and add punctation to exception messages.

Reply:
it's done !